src/Bicep.Core.IntegrationTests/ExtensibilityTests.cs

```diff
@@ -105,14 +105,15 @@ public void Storage_import_basic_test_loops_and_referencing()
 }]
 
 output sourceContainerName string = container.name
-output sourceContainerNameSquare string = container['name']
+var propName = 'name'
+output sourceContainerNameSquare string = container[propName]
 output miscBlobContainerName string = blobs[13 % 10].containerName
 output containerName string = blobs[5].containerName
-output base64Content string = blobs[3]['base64Content']
+output base64Content string = blobs[3].base64Content
 ");
             result.Should().NotHaveAnyDiagnostics();
             result.Template.Should().HaveValueAtPath("$.outputs['sourceContainerName'].value", "[reference('container').name]");
-            result.Template.Should().HaveValueAtPath("$.outputs['sourceContainerNameSquare'].value", "[reference('container').name]");
+            result.Template.Should().HaveValueAtPath("$.outputs['sourceContainerNameSquare'].value", "[reference('container')[variables('propName')]]");
```

Comment:
There seem to be a lot of test changes.  What's the purpose?  Is it to avoid dealing with linter errors?  Are you trying to add diversity to the expressions?  I'll have to look more closely to, but my initial reaction is to be hesitant to make too many test changes since I don't know what scenarios they're needing to test.  Thoughts?

Reply:
This test in particular is ensuring that property and array access of attributes are translated into the correct ARM JSON output, but the new linter will raise a diagnostic if array access is used when property access would have been permitted. I can update the `result.Should().NotHaveAnyDiagnostics()` instead, though that may result in more lines changed overall.

The other test changes were included because the scenario tests do not handle cases where a line emits two code fix suggestions, so lines like `runtimeValidRes1['sku']['name']` would require an update to the test harness to pass.

Reply:
Thanks, makes sense.  In that case, though, I would prefer:
1) for baseline tests, don't modify existing bicep test code without a good reason (feel free to add code if it makes sense) - instead, allow the new diagnostic to show up in the baseline.  If you need to keep the diagnostic from firing for whatever reason, you could use a #disable-next-line statement
2) for other tests, in general my recommendation for most scenarios is to add use ExcludingLinterDiagnostics to isolate the test from new linter rules being added:
```
   result.ExcludingLinterDiagnostics().Should().HaveDiagnostics(new[] {
```
You could also use #disable-next-line or modify the HaveDiagnostics, but my philosophy is that non-baseline tests ought to be stable against most linter rule additions (assumes the linter rule has its own set of tests).